src/assign.c

```diff
@@ -1040,6 +1040,14 @@ SEXP allocNAVector(SEXPTYPE type, R_len_t n)
   UNPROTECT(1);
   return(v);
 }
+// #3723 -- build NA vector like x -- mainly used to copy attributes pass this to writeNA
+SEXP allocNAVectorLike(SEXP x, R_len_t n) {
```

Comment:
why not put that in `assign.c`?
```c
SEXP v = PROTECT(allocNAVector(TYPEOF(thisvalue), nrow));
copyMostAttrib(thisvalue, v);
SET_VECTOR_ELT(dt, coln, newcol=v)
UNPROTECT(1);
```

Reply:
ok, I read `writeNA` and now I understand

Reply:
Although I am not sure if it make sense to copy attrs every time, some of them might be heavy.
Maybe check for class and redirect for int64, or copy class only?

Reply:
Feels weird to me to be so `int64`-specific in a fix... users with heavy `attr` can use `as.vector` to strip attributes?